manager/integration/tests/test_orphan.py

```diff
@@ -0,0 +1,940 @@
+        orphans = client.list_orphan()
+        for orphan in orphans:
+            delete_orphan(client, orphan.name)
+
+        with pytest.raises(subprocess.CalledProcessError):
+            exec_nsenter("ls {}".format(path))
+
+        # Check orphan CRs
+        count = wait_for_orphan_count(client, 0, 180)
+        assert count == 0
+
+    cleanup_volume(client, volume_name)
+    cleanup_host_disk(client, 'vol-disk-1')
+
+
+@pytest.mark.orphan
+def test_orphan_creation_and_deletion_in_multiple_disks(client, volume_name):  # NOQA
+    """
+    Test orphan creation and deletion in multiple disks
+    1. Create multiple new-disks for holding orphaned replica directories
+    2. Create a volume and attach to the current node
+    3. Create multiple orphaned replica directories by copying the active
+       replica directory
+    4. Verify orphan list contains the orphan CRs for replica
+       directories
+    5. Delete the orphaned replica directories
+    6. Verify orphan list is empty
+    """
+
+    lht_hostId = get_self_host_id()
+    node = client.by_id_node(lht_hostId)
+    disks = node.disks
+
+    # test delete disk exception
+    with pytest.raises(Exception) as e:
+        node.diskUpdate(disks={})
+    assert "disable the disk" in str(e.value)
+
+    # create multiple disks for node
+    disk_paths = []
+    node = client.by_id_node(lht_hostId)
+    disks = node.disks
+    path = create_host_disk(client, 'vol-disk-1',
+                            str(Gi), lht_hostId)
+    disk1 = {"path": path, "allowScheduling": True}
+    disk_paths.append(path)
+
+    path = create_host_disk(client, 'vol-disk-2',
+                            str(Gi), lht_hostId)
+    disk2 = {"path": path, "allowScheduling": True}
+    disk_paths.append(path)
+
+    update_disk = get_update_disks(disks)
+    # add new disk for node
+    update_disk["disk1"] = disk1
+    update_disk["disk2"] = disk2
+
+    # save disks to node
+    node = node.diskUpdate(disks=update_disk)
+    node = wait_for_disk_update(client, lht_hostId,
+                                len(update_disk))
+
+    # Create a volume
+    nodes = client.list_node()
+    volume = create_and_check_volume(client, volume_name, len(nodes), SIZE)
+    volume.attach(hostId=lht_hostId, disableFrontend=False)
+    wait_for_volume_healthy(client, volume_name)
+
+    # Create orphaned replica directories
+    orphaned_directories = []
+    num_orphans = 5
+    for replica in volume.replicas:
+        if replica.hostId != lht_hostId:
+            continue
+
+        for _ in range(num_orphans):
+            for disk_path in disk_paths:
+                path = os.path.join(disk_path, "replicas",
+                                    volume_name + "-" + generate_random_id(8))
+                orphaned_directories.append(path)
+                exec_nsenter("cp -a {} {}".format(replica.dataPath, path))
+
+    cleanup_volume(client, volume_name)
+
+    # Check orphan CRs
+    count = wait_for_orphan_count(client,
```

Comment:
`wait_for_orphan_count` actually does the number check w/o assertion, so probably we can enhance this to provide more reusable assertions instead of having another following general assert to do the duplicated check again. 

However, It's fine for now and leave to the later test cases refactoring.

Reply:
cc @longhorn/qa try to consider introducing assertion util functions in the future.